modules/tinymce/src/plugins/table/main/ts/ui/Buttons.ts

```diff
@@ -165,6 +169,303 @@ const addButtons = (editor: Editor, selectionTargets: SelectionTargets, clipboar
+    fetch: (callback) => {
+      callback([
+        {
+          type: 'menuitem',
+          text: 'Cut column',
+          icon: 'cut-column',
+          onAction: () => {
+            editor.execCommand('mceTableCutCol');
+          }
+        },
+        {
+          type: 'menuitem',
+          text: 'Copy column',
+          icon: 'duplicate-column',
+          onAction: () => {
+            editor.execCommand('mceTableCopyCol');
+          }
+        },
+        {
+          type: 'togglemenuitem',
+          text: 'Paste column before',
+          icon: 'paste-column-before',
+          onAction: () => {
+            editor.execCommand('mceTablePasteColBefore');
+          },
+          onSetup: (buttonApi) => {
+            buttonApi.setDisabled(editor.plugins.table.getClipboardCols().length === 0);
+          }
+        },
+        {
+          type: 'togglemenuitem',
+          text: 'Paste column after',
+          icon: 'paste-column-after',
+          onAction: () => {
+            editor.execCommand('mceTablePasteColAfter');
+          },
+          onSetup: (buttonApi) => {
+            buttonApi.setDisabled(editor.plugins.table.getClipboardCols().length === 0);
+          }
+        },
+      ]);
+    }
+  });
+
+  const tableClassList = getTableClassList(editor);
+
+  if (tableClassList.length === 0) {
+    // eslint-disable-next-line no-console
+    console.error('Missing table class list');
+  } else {
+    editor.ui.registry.addMenuButton('tableclass', {
+      icon: 'table-classes',
+      tooltip: 'Table styles',
+      fetch: (callback) => {
+        callback(Arr.map(tableClassList, (value) => {
+          const item: Menu.ToggleMenuItemSpec = {
+            text: value.title,
+            type: 'togglemenuitem',
+            onAction: (_api: Menu.MenuItemInstanceApi) => {
+              editor.execCommand('mceTableToggleClass', false, value.value);
+            },
+            onSetup: onSetupClassToggle(editor, 'tableclass', value.value)
+          };
+
+          return item;
+        }));
+      },
+      onSetup: selectionTargets.onSetupTable
+    });
+  }
+
+  const tableCellClassList = getCellClassList(editor);
+
+  if (tableCellClassList.length !== 0) {
+    editor.ui.registry.addMenuButton('tablecellclass', {
+      icon: 'table-cell-classes',
+      tooltip: 'Cell styles',
+      fetch: (callback) => {
+        callback(Arr.map(tableCellClassList, (value) => {
+          const item: Menu.ToggleMenuItemSpec = {
+            text: value.title,
+            type: 'togglemenuitem',
+            onAction: (_api: Menu.MenuItemInstanceApi) => {
+              editor.execCommand('mceTableCellToggleClass', false, value.value);
+            },
+            onSetup: onSetupClassToggle(editor, 'tablecellclass', value.value)
+          };
+
+          return item;
+        }));
+      },
+      onSetup: selectionTargets.onSetupCellOrRow
+    });
+  }
+
+  const alignTableButtons = [
```

Comment:
These are meant to be the same as what's shown in the dialog IIUC. As such, please don't duplicate them and instead make them use a shared definition, otherwise if these need to be changed at any point it's highly likely one or the other will be missed.